### ISSUE TITLE
Wire meta-labeling inference into engine for signal gating

### ISSUE THREAD

**Issue by kumargu** [kumargu/OpenQuant#103]
## Context

Meta-labeling (PR #98) trains offline but isn't integrated into live signal flow. With AUC=0.548 (crypto) and 0.586 (tech), the model is weak but could still improve win rate by filtering the worst signals.

### Option A: Simple threshold gate (Python-side, quick)
In \`paper_trading/runner.py\`, after getting intents from the engine:
```python
meta_confidence = meta_model.predict_proba(meta_features)
if meta_confidence < 0.55:
    skip_trade()
```

### Option B: Decision tree export to Rust (better latency)
Export the gradient-boosted model as a simple decision tree (depth=3):
```python
from sklearn.tree import DecisionTreeClassifier
simple = DecisionTreeClassifier(max_depth=3)
simple.fit(X, model.predict(X))  # distill
# Export as if/else rules for Rust
```

### Option C: Feature importance-based hard rules (simplest, most robust)
From the meta-labeling results, the top features are:
1. signal_score > threshold
2. regime_change_prob < 0.3 (avoid trading during transitions)
3. relative_volume > 1.5 (require strong participation)
4. hour_of_day in [10, 11, 12, 13, 14] (avoid open/close)

These can be hardcoded as additional gates in the combiner without any ML model.

## Recommendation
Start with **Option C** — it's the most robust and doesn't add ML complexity to the hot path. The meta-labeling analysis identified the rules; we just need to implement them as explicit gates.
